### PR TITLE
Added unit square, react to mouse movements

### PR DESCRIPTION
**Pull request by elayabharath**
+ color and sorting to be done

The shrunk image doesn't show the hover behaviour properly, you may have to open the image in a new tab for better clarity.

Screenshot 2019-12-24 at 4 11 43 PM

### CODE REVIEW

File: lib/interactive.dart [hunk 1]
```diff
@@ -21,6 +21,10 @@ const XAXIS_CSS = "x-axis";
 const XAXIS_LABEL_CSS = "x-axis--label";
 const LEGEND_ITEM_CSS = "legend-item";
 
+// to do: make this configurable based on number of people
+const SQ_IN_ROW = 6;
+const SQ_SIZE = 12;
```
Comment (lukechurch) [inline lib/interactive.dart]: Consider 'SQ_WIDTH'

File: lib/interactive.dart [hunk 2]
```diff
@@ -194,9 +198,53 @@ class Interactive {
       peopleByLabel[key].add(people);
     });
 
+    for (var i = 0; i < xAxisCategories.length; ++i) {
+      var colSVG = svg.SvgElement.tag("g");
```
Comment (lukechurch) [inline lib/interactive.dart:251]: What does this do?

File: lib/interactive.dart [hunk 3]
```diff
@@ -194,9 +198,53 @@ class Interactive {
...
+
+      var colData = peopleByLabel[xAxisCategories[i].value];
+      num colOffset = (i + 0.5) * (chartWidth / xAxisCategories.length);
```
Comment (lukechurch) [inline lib/interactive.dart]: There's a lot of numbers floating around here - consider colOffsetPx

File: lib/interactive.dart [hunk 4]
```diff
@@ -194,9 +198,53 @@ class Interactive {
...
+
+      for (var j = 0; j < colData.length; ++j) {
+        num x = (j % SQ_IN_ROW - SQ_IN_ROW / 2) * SQ_SIZE + colOffset;
+        num y = (CHART_HEIGHT - CHART_XAXIS_HEIGHT - 1.5 * SQ_SIZE) -
```
Comment (lukechurch) [inline lib/interactive.dart]: truncate does the same as floor?

File: lib/interactive.dart [hunk 5]
```diff
@@ -194,9 +198,53 @@ class Interactive {
...
+        colSVG.append(sqGroup);
+      }
+
+      svgContainer.append(colSVG);
+    }
+
     return chartWrapper..append(svgContainer);
   }
 
+  void handleMouseEnter(svg.SvgElement rect, int w) {
```
Comment (lukechurch) [inline lib/interactive.dart]: what does `w` mean here? Could you expand to a longer variable name?

File: lib/interactive.dart [hunk 6]
```diff
@@ -194,9 +198,53 @@ class Interactive {
...
+    rect.parent.append(rect);
+    rect..setAttribute("transform", "translate(${-2 * w}, ${-2 * w}) scale(2)");
+  }
+
+  void handleMouseOut(svg.SvgElement rect) {
+    rect..setAttribute("transform", "translate(0, 0) scale(1)");
+  }
+
   html.DivElement _renderLegend() {
     var legendWrapper = html.DivElement()..classes = [BS_ROW_CSS];
```
Comment (lukechurch) [inline lib/interactive.dart]: What does `BS_ROW_CSS` mean?

**Review by lukechurch — APPROVED** [review]
Thanks @elayabharath - functionality LGTM - but I think the variable names could use being a bit more verbose to help people less familiar with the code

File: lib/interactive.dart (hunk 1 above)
Comment (elayabharath) [reply lib/interactive.dart]: Fixed

File: lib/interactive.dart (hunk 2 above)
Comment (elayabharath) [reply lib/interactive.dart:251]: This is a group where I add all squares of each column - a group so I can do background highlights potentially later

File: lib/interactive.dart (hunk 6 above)
Comment (elayabharath) [reply lib/interactive.dart]: Bootstrap row css - removed the prefix, fixed

File: lib/interactive.dart (hunk 5 above)
Comment (elayabharath) [reply lib/interactive.dart]: Expanded as sqWidth

File: lib/interactive.dart (hunk 4 above)
Comment (elayabharath) [reply lib/interactive.dart]: Seems to do the same i.e. drops the decimal digits, but i think floor is a lot more explicit. Fixed

File: lib/interactive.dart (hunk 3 above)
Comment (elayabharath) [reply lib/interactive.dart]: Fixed

**Pull request merged**